stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/WebSocketClientHandshakeListener.java

```diff
@@ -20,75 +20,60 @@
 
-import java.util.concurrent.CountDownLatch;
-
-import static org.ballerinalang.net.http.HttpConstants.PROTOCOL_HTTP_PKG_ID;
-
 /**
  * The handshake listener for the client.
  *
  * @since 0.983.1
  */
-public class WebSocketClientHandshakeListener implements ClientHandshakeListener {
+public abstract class WebSocketClientHandshakeListener implements ClientHandshakeListener {
 
     private final WebSocketService wsService;
-    private final WebSocketClientConnectorListener clientConnectorListener;
+    private final WebSocketClientListener clientConnectorListener;
     private final boolean readyOnConnect;
     private final ObjectValue webSocketClient;
-    private CountDownLatch countDownLatch;
 
     public WebSocketClientHandshakeListener(ObjectValue webSocketClient,
-                                            WebSocketService wsService,
-                                            WebSocketClientConnectorListener clientConnectorListener,
-                                            boolean readyOnConnect, CountDownLatch countDownLatch) {
+                                                WebSocketService wsService,
+                                                WebSocketClientListener clientConnectorListener,
+                                                boolean readyOnConnect) {
         this.webSocketClient = webSocketClient;
         this.wsService = wsService;
         this.clientConnectorListener = clientConnectorListener;
         this.readyOnConnect = readyOnConnect;
-        this.countDownLatch = countDownLatch;
     }
 
     @Override
     public void onSuccess(WebSocketConnection webSocketConnection, HttpCarbonResponse carbonResponse) {
         //using only one service endpoint in the client as there can be only one connection.
         webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD,
-                            HttpUtil.createResponseStruct(carbonResponse));
-        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_HTTP_PKG_ID,
-                                                                           WebSocketConstants.WEBSOCKET_CONNECTOR);
+                HttpUtil.createResponseStruct(carbonResponse));
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(HttpConstants.PROTOCOL_HTTP_PKG_ID,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
         WebSocketOpenConnectionInfo connectionInfo = getWebSocketOpenConnectionInfo(webSocketConnection,
-                                                                                    webSocketConnector);
+                webSocketConnector);
         WebSocketUtil.populateWebSocketCaller(webSocketConnection, webSocketClient);
         clientConnectorListener.setConnectionInfo(connectionInfo);
         if (readyOnConnect) {
-            WebSocketUtil.readFirstFrame(webSocketConnection, webSocketClient);
+            webSocketConnection.readNextFrame();
```

Comment:
Note to make sure readNextFrame is not called multiple times. If it is called in this method it will be called in the sub classes also and there is a chance of it being called multiple times. Can you be sure it is not multiple times?

Reply:
**Where are we calling the readFirstFrame function? There is some serious flaw in this implementation?**

Reply:
Can you try calling ready twice without the  retryConfig it will not return an error

Reply:
Sice you refactored the code, ready() function only called in the websocketClient. For that you changed the ready function's related things from webSocketConnector to webSocketClient.

https://github.com/ballerina-platform/ballerina-lang/pull/19219/files#diff-862cb563e30801c3a2ec38f6ac3b6266R20
https://github.com/ballerina-platform/ballerina-lang/pull/19219/files#diff-862cb563e30801c3a2ec38f6ac3b6266R98
https://github.com/ballerina-platform/ballerina-lang/pull/19219/files#diff-7e9dc1216740c92a41aaaba2faf0bc77R42
https://github.com/ballerina-platform/ballerina-lang/pull/19219/files#diff-c344ae55fd805ed9fdc231708f0e342fR73

But now, ready() function is using in the both client(FailoverWebSocketClient & websocketClient) .  So we need to use the ready function's related things in the  common client.that means, We need to implement the ready function in the websocketConnector.
As per the current implementation, I followed the old implementation. 

But the line[1] depends on those changes.So I changed that line to older[2] and checked the samples and testcases.Now, all are worked successfully.

[1] https://github.com/ballerina-platform/ballerina-lang/pull/19054/files#diff-c344ae55fd805ed9fdc231708f0e342fR67
[2] https://github.com/ballerina-platform/ballerina-lang/pull/19219/files#diff-c344ae55fd805ed9fdc231708f0e342fR73